### ISSUE TITLE
Automatically insert import on code completion

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#21931]
The code completion suggestion list sometimes contains names that are not yet imported into the current scope. When that happens, the import for the name should be automatically added when the suggestion is selected.

**Comment by alexander-doroshko** [comment]
We have this functionality in IntelliJ IDEA already and we are loosing it as we are going to switch to DAS driven code completion. It is not always adding a new import, it may be updating show/hide of already existing import (if such suggestions are shown at all).

**Comment by alexander-doroshko** [comment]
> sometimes contains names that are not yet imported

Just curious in which cases non-imported names are suggested in completion and in which - not?

**Comment by danrubel** [comment]
We no longer suggest any types that are not already imported. This is a result of my refactoring the code completion engine in Nov/Dec 2015 so that it consistently used resolved types throughout. I did not include non-imported types when I refactored the code because the implementation at that time was flawed and there is a blocking issue that must be tackled first.

See https://github.com/dart-lang/sdk/issues/25820 for more information.
